tests/test_gui.py

```diff
@@ -0,0 +1,1152 @@
+
+
+def test_auto_elide_label(app_dialog):
+    # Click on the Context Selector widget
+    app_dialog.open_demo_pane("Auto-Elide Label")
+    assert app_dialog.root.captions[
+        "Auto-Elide Label"
+    ].exists(), "Not on the Auto-Elide Label Widget"
+
+    # Validate default lable value
+    assert app_dialog.root.captions[
+        "Lorem ipsum dolor sit amet, consectetur adipiscing el..."
+    ].exists(), "Default Auto-Elide text value is good"
+
+    # Slide to the left side to remove the text and validate
+    labelIndicator = first(app_dialog.root.buttons["Page left"])
+    width, height = labelIndicator.size
+    app_dialog.root.indicators.Position[1].mouseSlide()
+    labelIndicator.mouseDrag(width * 0, height * 0)
+    assert app_dialog.root.captions[""].exists(), "Empty Auto-Elide text value is good"
+
+    # Slide to the right side to show all the text and validate
+    labelIndicator = first(app_dialog.root.buttons["Page right"])
+    width, height = labelIndicator.size
+    app_dialog.root.indicators.Position[1].mouseSlide()
+    labelIndicator.mouseDrag(width * 1, height * 0)
+    assert app_dialog.root.captions[
+        "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque non posuere lorem. Donec non lobortis mauris...."
+    ].exists(), "Full Auto-Elide text value is good"
+
+
+def test_global_search(app_dialog):
+    # Click on the Global Search widget
+    app_dialog.open_demo_pane("Global Search")
+    assert app_dialog.root.captions[
+        "Global Search"
+    ].exists(), "Not on the Global Search Widget"
+
+    # Search for a specific entity
+    app_dialog.root.textfields.typeIn("Art")
+    topwindows.listitems["Art"].get().mouseClick()
+
+    # Validate search complete successfully
+    assert app_dialog.root.captions[
+        "Task 'Art' with id 448 activated"
+    ].exists(), "Global search is not working"
+
+
+def test_help_screen(app_dialog):
+    # Click on the Help Screen Popup widget
+    app_dialog.open_demo_pane("Help Screen Popup")
+    assert app_dialog.root.captions[
+        "Help Screen Popup"
+    ].exists(), "Not on the Help Screen Popup Widget"
+
+    # Click on the show_help_screen button
+    app_dialog.root.buttons["show_help_screen()"].get().mouseClick()
+
+    # Validate Show Help Screen
+    assert app_dialog.root.dialogs[
+        "Toolkit Help"
+    ].exists(), "Show Help Screen is not showing up"
+    assert (
+        app_dialog.root.dialogs["Toolkit Help"]
+        .buttons["Jump to Documentation"]
+        .exists()
+    ), "Jump to Documentation button is not available"
+    assert (
+        app_dialog.root.dialogs["Toolkit Help"].buttons["Close"].exists()
+    ), "Close button is not available"
+    assert (
+        app_dialog.root.dialogs["Toolkit Help"]
+        .buttons["Scroll to the next slide"]
+        .exists()
+    ), "Scroll to the next slide button is not available"
+
+    # Click on Scroll to the next slide until you reach the last slide
+    while (
```

Comment:
Never have potential endless loops in a test that could hang in CI. Loop for the expected number of times and then make sure the GUI is in the right state.